confd/pkg/backends/calico/client.go

```diff
@@ -1694,6 +1706,21 @@ func (c *client) DeleteStaticRoutes(cidrs []string) {
 	c.onNewUpdates()
 }
 
+// StaticRouteExists check if static route exists
+func (c *client) StaticRouteExists(cidr string) bool {
```

Comment:
This all depends on the answer to my question on routes.go line 310, though, which I haven't fully wrapped my head around yet.

Reply:
Based on my answer for line 310. For me, the most clean way of doing this is to do it from the route generator. Since addRoutesLockHeld is non-exported you basically have to do it through AddStaticRotues().

Calling AddStaticRoutes() in all cases would bump cache revision for all services changes even though there are no real change, and I guess that is sub-optimal?